### ISSUE TITLE
[Feature Request] Allow "Group by Origin" to sort Features not granted by a specific source, such as manually-added features

### ISSUE THREAD

**Issue by TremendoDude** [foundryvtt/dnd5e#2837]
As of 3.0.0, the new Default 5e Character Sheet now allows players to Group by Origin, separating features by Class, Race, Background, and Other. However, at the moment, it only groups features that were actively granted by the listed Class/Race/Background through automated granting of features.

There are many situations where someone will wish to add a custom feature or swap out a feature in accordance with these categories without having to go back through the automated feature-granting stages. Beyond that, when using a source that needs to be recreated by hand (such as a Race or Class from a physical sourcebook), the current Group by Origin option is unusable unless the source is first recreated with all automation intact, which is a shame when it'd often be so much easier to just create the features as needed.

It would be great if it were possible to Group by Origin in a less strict way, to allow making use of this useful sorting method with features not granted through automated recreated sources.
- One idea would be to allow features to group in accordance with Feature Type. Race Features would be grouped along with features granted by the Race, Class features would be grouped along with features granted by the class (perhaps with manual sorting allowing between classes in Multiclass situations), etc.
- Another idea would be to allow manual sorting between the Class, Race, Background, and other lists. By default features would appear in the list appropriate to their source, but players would be able to drag and drop a feature from one list to another. So for example, a manually added Class feature, which would by default appear in "Other Features", could be then dragged manually to the Class Features list.

**Comment by etiquettestartshere** [comment]
I agree. Its particularly frustrating because some features that may be added after the fact, as the story or play dictates, often make the most sense logically to be added at level 1, since they aren't mean to represent something that was gained by level up, but just be part of that category, and level 1 tends to have the most steps of Advancement that must be replayed when selecting Modify Choices. The UX for adding even one new feature to a class category at level 1, for instance, is pretty grueling.

If drag and drop is too difficult, another option would be to have the feature groups be populated on items after they are already on the actor, similar to how items will populate resource consumption after the item is added to an actor.

**Comment by M0nk3yy** [comment]
I would definitely be in favour of this. Upon first encountering such a situation, my default assumptions were that I'd either be able to: 

1. Drag and drop them into the section I wanted if they were not correctly categorized. 

or

2. Go to the details tab and change something near the Class Feature drop-down. Such as potentially a second drop-down that shows classes currently on the character sheet. 


I would personally encounter such a situation fairly regularly, as I don't have all the class and subclass features/choices prepared ahead of time. Especially when there are multiple options and I don't know what is going to be picked, since I'd only prepare what the player actually needs. Whatever the solution, I do think some sort of manual sort would be a nice workflow improvement.

**Comment by itteerde** [comment]
I agree that some way of manually fixing it through UI is needed. There are plenty good reasons to copy customized versions of features even before getting into adding Item-Macro and such.

**Comment by PhDaveMil** [comment]
For whatever it is worth, I would personally find it extremely helpful if I could manually choose "call, race/species, other" for each feature on a character's sheet for the purpose of manually grouping them. That feature would make it much easier to fix sheets that for, whatever reason, are not grouping correctly. It would make the D&D 5e system infinitely more pleasant to use for me.

**Comment by AstroQuantico** [comment]
I also believe it is an essential function. Many automation modules end up moving the Features to "Other Feature" area, and to return them to their original area, you practically have to redo the entire character sheet. Sometimes I would like to create new groups of features and items to organize better the character sheet.

**Comment by MichaelAlbert98** [comment]
I've personally run into this issue in every 5e game I am running/participating it. The granularity of Group by Origin is much better than having them listed as passive/active effects, but shoving any custom abilities into the Other Features section at the bottom of the list is very frustrating. Especially since those tend to be the more complex/fiddly features that need to be referenced more often than something like Darkvision or Movement Increase.

I would much prefer the ability to manually drag and drop features from one section to another in order to customize the feature tab or if that isn't tenable, the ability for features to be moved to their respective category when selecting a feature type e.g. Class/Background/Race

**Comment by w15p** [comment]
_repost of my comment on 2783..._
I'd really like to see this change - it seems strange to me that you cannot create the same character sheet manually that you can with "character advancement" - there are often going to be edge cases that are not handled via character advancement and having features shunted into "Other Features" when they are obviously Class Features is problematic. Additionally being able to specify which Class in the case of Multi-Classed characters is another important consideration.
